dbt_invoke/properties.py

```diff
@@ -502,13 +503,22 @@ def _get_columns(ctx, resource_location, resource_dict, **kwargs):
         result_lines = _utils.dbt_run_operation(
             ctx, _MACRO_NAME, hide=True, logger=_LOGGER, sql=sql, **kwargs
         )
-    result_list_strings = [
-        s for s in result_lines if s.startswith('[') and s.endswith(']')
-    ]
-    # Take the last line that started with '[' and ended with ']'
-    # (just in case there is multi-line output)
-    columns_string = result_list_strings[-1].strip('][').split(', ')
-    columns = [col.strip("'") for col in columns_string]
+
+    # from dbt-core v1.0.0 onwards, run_operations INFO logs have code M011 and messages have key 'msg'
+    # https://github.com/dbt-labs/dbt-core/blob/22b1a09aa218e8152b0c2dd261abe2503ea15ddb/core/dbt/events/types.py#L401
+    relevant_line = list(filter(lambda x: x.get('code') == 'M011', result_lines))
+
+    if len(relevant_line) == 1:
+        columns = relevant_line[0].get('msg')
+
+    else:
+        # for older dbt-core versions, we need to cross fingers a little harder
+        relevant_line = result_lines[1:]
+        # also, the message key is different
+        columns = relevant_line[0].get('message')
+        # columns are not passed as valid json but as a string representation of a list
+        columns = ast.literal_eval(columns)
+
```

Comment:
I think this is a good design in general.  I have a few very minor suggestions I would like to get your thoughts on:

- In the unlikely case where `len(relevant_line)` is greater than 1, I think we would still want to remain within the first branch of the if statement.
- Unless there is a reason to change it, I think we should continue to grab the last relevant line (not that I had any really strong reasoning for the last line vs. first line when I originally wrote it)
- Change the name `relevant_line` to `relevant_lines` because it is a list that could theoretically contain multiple lines, even though it will be just one line most of the time.

```suggestion
    relevant_lines = list(filter(lambda x: x.get('code') == 'M011', result_lines))
    if len(relevant_lines) >= 1:
        columns = relevant_lines[-1].get('msg')
    else:
        # for older dbt-core versions, we need to cross fingers a little harder
        relevant_lines = result_lines[1:]
        # also, the message key is different
        columns = relevant_lines[-1].get('message')
        # columns are not passed as valid json but as a string representation of a list
        columns = ast.literal_eval(columns)
```

Reply:
I agree, especially as we have "control" over the macro. I doesn't feel like the most elegant solution but I also don't have anything better to offer atm..